### PR TITLE
[CELEBORN-348] Support fetchTime in load-aware slots assignment strategy

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
Currently load-aware slots assignment strategy only considers flush time, but flush time might not correctly
reflects disk performance because we do not flush to disk. It's more reasonable to also consider fetch time,
which more precisely reflects disk perf.

This PR considers both flush and fetch time, and uses ```celeborn.slots.assign.loadAware.flushTimeWeight```
and ```celeborn.slots.assign.loadAware.fetchTimeWeight``` for flush/fetch time weight respectively.

### Why are the changes needed?
As above.

### Does this PR introduce _any_ user-facing change?
Yes, adds two more configs: ```celeborn.slots.assign.loadAware.flushTimeWeight```
and ```celeborn.slots.assign.loadAware.fetchTimeWeight```.

### How was this patch tested?
UT and regression.

### CODE REVIEW

**Comment by waitinfuture** [comment]
I have run regression and results is as expected. My Celeborn cluster is heterogeneous, 6 of them have better disks and 4 of them have poorer disks. The following snapshot is slots allocated per worker, and we can see the workers with better disks are allocated more slots than poor ones as time goes.
image

**Comment by waitinfuture** [comment]
ping @FMX

File: common/src/main/scala/org/apache/celeborn/common/meta/TimeWindow.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.meta
+
+import java.util.concurrent.atomic.LongAdder
+
+class TimeWindow(windowSize: Int, minWindowCount: Int) {
```
Comment (FMX) [inline common/src/main/scala/org/apache/celeborn/common/meta/TimeWindow.scala:22]: IMO there is a weak connection with time. Maybe naming it LimitedSlidingWindowReservoir would be better.

**Review by FMX — APPROVED** [review]
LGTM.

**Pull request merged**